pkg/appmgmt/general/podevent_handler_test.go

```diff
@@ -73,21 +73,48 @@ func TestHandleAsyncEventWhenNotRecovering(t *testing.T) {
 func TestRecoveryDone(t *testing.T) {
 	amProtocol := cache.NewMockedAMProtocol()
 	podEventHandler := NewPodEventHandler(amProtocol, true)
+
 	pod1 := newPod("pod1")
+	pod2 := newPod("pod2")
+	pod3 := newPod("pod3")
+	pod3v1 := newPodWithResourceVersion(pod3.GetName(), "1")
+	pod3v2 := newPodWithResourceVersion(pod3.GetName(), "2")
+
 	podEventHandler.HandleEvent(AddPod, Informers, pod1)
+	podEventHandler.HandleEvent(AddPod, Informers, pod2)
 	podEventHandler.HandleEvent(DeletePod, Informers, pod1)
+	podEventHandler.HandleEvent(AddPod, Informers, pod3v1)
+	podEventHandler.HandleEvent(UpdatePod, Informers, pod3v2)
 
-	podEventHandler.RecoveryDone()
+	seenEvents := map[types.UID]string{
+		pod2.UID: pod2.GetResourceVersion(), // should not be added
+		pod3.UID: pod3.GetResourceVersion(), // should be updated with new version and ultimately completed
+	}
+	podEventHandler.RecoveryDone(seenEvents)
 
 	assert.Equal(t, len(podEventHandler.asyncEvents), 0)
 	app := amProtocol.GetApplication(appID)
+
 	task, err := app.GetTask("pod1")
 	assert.NilError(t, err)
 	assert.Equal(t, cache.TaskStates().Completed, task.GetTaskState())
+
+	_, err = app.GetTask("pod2")
+	assert.ErrorContains(t, err, "task pod2 doesn't exist in application")
+
+	task, err = app.GetTask("pod3")
+	assert.NilError(t, err)
+	assert.Equal(t, "1", task.GetTaskPod().GetResourceVersion())
```

Comment:
I'd like to hear feedback on this assertion specifically, as the `UpdatePod` event simply completes the task without actually updating the associated pod. So the tests are passing but this is not extremely intuitive.

Reply:
I think the assumption in the code that a UpdatePod event should transition the pod to finished is not correct. In the handling part of `Manager.updatePod` a check of the current pod and new pod is performed. Then it only transitions the pod if needed by calling the handler.

With VPA (vertical pod auto scaler) there might be more that we need to handle for updates, that is a discussion for later.

I think `PodEventHandler.updatePod` needs to get both pods and the logic of the `Manager.updatePod` needs to move to it. That needs a bit of "surgery" of the PodEventHandler, **BUT** not for as part of this change.

Reply:
Created an umbrella (https://issues.apache.org/jira/browse/YUNIKORN-1765) and a subtask (https://issues.apache.org/jira/browse/YUNIKORN-1766) for this.